cli_tools_tests/e2e/assert.go

```diff
@@ -12,17 +12,16 @@
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
-// Package assert contains e2e tests assertion functions
-package assert
+package e2e
```

Comment:
I don't see anything CLI tools e2e specific in here. Do you think this file can go to e.g. common/testing?

Reply:
The assertions here are highly coupled to the e2e tests since they use the custom `junitxml` testcase that is only used in the e2e tests.

Reply:
These functions are pretty generic, I don't see anything E2E/module test specific in them. What do you think about moving them to `common/testing`?

Reply:
`go/e2e_test_utils/junitxml` is a legacy way of formatting golang test results for prow, and therefore I'd like to keep it isolated to where it's used. Otherwise, an unsuspecting dev in the future will try to use it from `cli_tools` or `cli_tools_tests/modules` and find themselves in a pickle.

Reply:
> legacy way of formatting golang test results for prow

What do you mean by legacy here? What is the current way of doing this?

Reply:
By legacy, I mean that this code's usecase became popular, so its functionality has been absorbed into the Prow ecosystem:
* https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md
* https://github.com/jstemmer/go-junit-report

For new projects, the current approach is to use the [gotest](https://github.com/GoogleCloudPlatform/guest-test-infra/tree/master/container_images/gotest) runner. It checks out a repo, executes `go test` against a directory, and formats and uploads the results. The project owner focuses on writing tests using Go's test framework, and the platform ensures it works with testgrid and spyglass.

In addition to gotest, [gointegtest](https://github.com/GoogleCloudPlatform/guest-test-infra/pull/37) is new, and it executes tests across a parallel fleet of workers.

Reply:
It seems we can consider refactoring our tests if there's a reason to do so, or at least consider the new framework for new tests.

Regarding assert.go I still think that the new `common` package is a more appropriate spot, but I won't block this PR just because of that.